math-scala/src/main/scala/org/apache/mahout/math/algorithms/regression/OrdinaryLeastSquaresModel.scala

```diff
@@ -0,0 +1,123 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one
+  * or more contributor license agreements. See the NOTICE file
+  * distributed with this work for additional information
+  * regarding copyright ownership. The ASF licenses this file
+  * to you under the Apache License, Version 2.0 (the
+  * "License"); you may not use this file except in compliance
+  * with the License. You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing,
+  * software distributed under the License is distributed on an
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  * KIND, either express or implied. See the License for the
+  * specific language governing permissions and limitations
+  * under the License.
+  */
+
+package org.apache.mahout.math.algorithms.regression
+
+import org.apache.mahout.math.{Vector => MahoutVector}
+import org.apache.mahout.math.drm.RLikeDrmOps._
+import org.apache.mahout.math.drm.DrmLike
+import org.apache.mahout.math.scalabindings.{dvec, _}
+import org.apache.mahout.math.scalabindings.RLikeOps._
+
+import scala.reflect.ClassTag
+
+/**
+  * import org.apache.mahout.math.algorithms.regression.OrdinaryLeastSquares
+  * val model = new OrdinaryLeastSquares()
+  *
+  * model.calcStandardErrors = true
+  * 
+  */
+class OrdinaryLeastSquaresModel[K]
+  extends LinearRegressorModel[K] {
+  // https://en.wikipedia.org/wiki/Ordinary_least_squares
+
+
+  def predict(drmPredictors: DrmLike[K]): DrmLike[K] = {
+    drmPredictors %*% beta
+  }
+
+}
+
+class OrdinaryLeastSquares[K] extends LinearRegressorModelFactory[K] {
+
+  var addIntercept: Boolean = _
+  var calcStandardErrors: Boolean = _
+
+  def setHyperparameters(hyperparameters: Map[Symbol, Any] = Map('foo -> None)): Unit = {
+    calcStandardErrors = hyperparameters.asInstanceOf[Map[Symbol, Boolean]].getOrElse('calcStandardErrors, true)
+    addIntercept = hyperparameters.asInstanceOf[Map[Symbol, Boolean]].getOrElse('addIntercept, true)
+  }
+
+  def fit(drmFeatures: DrmLike[K],
+          drmTarget: DrmLike[K],
+          hyperparameters: (Symbol, Any)*): OrdinaryLeastSquaresModel[K] = {
+
+    val model = new OrdinaryLeastSquaresModel[K]()
+    if (hyperparameters != Map("" -> None)) {
+      // not sure this is right...
+      setHyperparameters(hyperparameters.toMap)
+    }
+
+    model.drmY = drmTarget
+    if (drmFeatures.nrow != drmTarget.nrow){
+      throw new Exception(s"${drmFeatures.nrow} observations in features, ${drmTarget.nrow} observations in target, must be equal.")
+    }
+
+    var X = drmFeatures
+    if (addIntercept) {
+      X = X cbind 1
+    }
+    val drmXtXinv = solve(X.t %*% X)
```

Comment:
Do we have any around pulling `X.t %*% X` into memory here?  I thought that recently I heard that this was being done differently here.